### ISSUE TITLE
Fix clang in gentoo

### ISSUE THREAD

**Issue by jbakosi** [quinoacomputing/buildenv#5]
After gentoo updated its default gcc from 4 to 5, linking to Kokkos code in Trilinos fails with a link error using the clang compiler only:
```
../LinSys/libquinoa_linsys.so: undefined reference to `Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::Serial, Kokkos::HostSpace>::name[abi:cxx11]()'
```
This is most likely due to gcc defaulting to a new C++11 ABI starting with gcc 5. More details at https://wiki.gentoo.org/wiki/Upgrading_from_gcc-4.x_to_gcc-5.x.

Attempting to switch to clang 4 using
```
diff --git a/gentoo b/gentoo
index 50d763e..906fe11 100644
--- a/gentoo
+++ b/gentoo
@@ -32,6 +32,13 @@ RUN flaggie virtual/mpi +romio
 RUN flaggie sci-libs/hdf5 +mpi
 RUN flaggie sys-cluster/openmpi +romio
 RUN flaggie sys-devel/llvm +clang
+RUN flaggie sys-devel/llvm-4* +~amd64
+RUN flaggie =sys-libs/compiler-rt-sanitizers-4.0.0 +~amd64
+RUN flaggie =sys-devel/clang-runtime-4.0.0 +~amd64
+RUN flaggie =app-vim/llvm-vim-4.0.0 +~amd64
+RUN flaggie =sys-libs/compiler-rt-4.0.0 +~amd64
+RUN flaggie =sys-libs/libomp-4.0.0 +~amd64
+RUN flaggie sys-devel/clang-4* +~amd64
 RUN emerge -1 clang
 RUN emerge -1 boost sys-cluster/charm
 RUN emerge doxygen
@@ -43,6 +50,6 @@ RUN emerge --onlydeps quinoa
 RUN groupadd -r quinoa
 RUN useradd -r -m -g quinoa quinoa
 USER quinoa
-ENV PATH=/usr/lib64/ccache/bin:${PATH}${PATH:+:}
+ENV PATH=/usr/lib64/ccache/bin:/usr/lib/llvm/4/bin:${PATH}${PATH:+:}
 WORKDIR /home/quinoa
 RUN pip install --user codecov coverxygen
```
yields the same link problem.

Possible options for a fix:
- build Trilinos with clang, keep everything else built with gcc
- build all our TPLs with clang
- build the whole gentoo stack with clang

Once this is done we need to re-enable the clang builds in CI, by reverting https://github.com/quinoacomputing/quinoa/commit/965981da.

**Comment by junghans** [comment]
That looks similar: http://stackoverflow.com/questions/33368237/undefined-reference-in-clang-when-linking-to-a-library-compiled-with-gcc5

and here is how to use clang inside emerge https://wiki.gentoo.org/wiki/Clang

**Comment by junghans** [comment]
This has been fixed by compiling trilinos with clang: b0633cb47ce31ea3687c35f13543f7f9d2692c10
